server/users.ts

```diff
@@ -1040,12 +1040,19 @@ export class User extends Chat.MessageContext {
 			Rooms.get(roomid)!.onLeave(oldUser);
 		}
 
-		if (this.locked === '#dnsbl' && !oldUser.locked) this.locked = null;
-		if (!this.locked && oldUser.locked === '#dnsbl') oldUser.locked = null;
+		const startLock = this.locked;
```

Comment:
These should be `oldLocked` and `oldSemilocked` to be consistent with the rest of our code.